java/ql/src/Security/CWE/CWE-022/TaintedPath.ql

```diff
@@ -41,10 +39,8 @@ class TaintedPathConfig extends TaintTracking::Configuration {
 
   override predicate isSanitizer(DataFlow::Node node) {
     exists(Type t | t = node.getType() | t instanceof BoxedType or t instanceof PrimitiveType)
-  }
-
-  override predicate isSanitizerGuard(DataFlow::BarrierGuard guard) {
-    guard instanceof ContainsDotDotSanitizer
+    or
+    node = DataFlow::BarrierGuard<containsDotDotSanitizer/3>::getABarrierNode()
```

Comment:
Can you elaborate, why you can basically turn the isSanitizer and isSanitizerGuard predicate into a disjunkt in isSanitizer?

Reply:
That's exactly how BarrierGuards are currently implemented by the library.